vsphere/resource_vsphere_compute_cluster.go

```diff
@@ -1455,6 +1505,60 @@ func expandVsanDatastoreConfig(d *schema.ResourceData, meta interface{}) (*vsant
 	return conf, nil
 }
 
+func buildVsanStretchedClusterReq(d *schema.ResourceData, cluster types.ManagedObjectReference) (*vsantypes.VSANVcConvertToStretchedCluster, error) {
+	log.Printf("[DEBUG] building vsan stretched cluster request...")
+	conf := d.Get("vsan_stretched_cluster_conf").(*schema.Set).List()[0].(map[string]interface{})
+
+	if conf["preferred_fault_domain_host_ids"] == "" {
```

Comment:
I'm not sure how this runs without panicking, this is a TypeSet so needs to go through with the conversions. I think a simpler design though is to make it `Required: true` and not `Optional` that should enforce that the Set is non-empty (at least I think so anyways, can easily be verified).